core/mise/mise.go

```diff
@@ -120,6 +121,55 @@ func (m *Mise) GetAllVersions(pkg, version string) ([]string, error) {
 	return versions, nil
 }
 
+// gets all package versions from mise that are defined in the app directory environment
+// this can include additional packages defined outside the app directory, but we filter those out
+func (m *Mise) GetPackageVersions(ctx MiseAppContext) (map[string]*MisePackageInfo, error) {
+	appDir := ctx.GetAppSource()
+	output, err := m.runCmd("--cd", appDir, "list", "--current", "--json")
```

Comment:
I think we should run in [paranoid](https://mise.jdx.dev/paranoid.html) mode. I am a bit worried about executing any user code here since that is by definition untrusted.